mesonbuild/mdist.py

```diff
@@ -178,33 +181,44 @@ def check_dist(packagename, meson_command, privdir):
     print('Distribution package %s tested' % packagename)
     return 0
 
-def run(args):
-    src_root = args[0]
-    bld_root = args[1]
-    meson_command = args[2:]
+def determine_archives_to_generate(options):
+    result = []
+    for i in options.formats.split(','):
+        if i not in archive_choices:
+            sys.exit('Value "{}" not one of permitted values {}.'.format(i, archive_choices))
+        result.append(i)
+    if len(i) == 0:
+        sys.exit('No archive types specified.')
+    return result
+
+def run(options):
+    b = build.load('.')
+    # This import must be load delayed, otherwise it will get the default
+    # value of None.
+    from mesonbuild.mesonlib import meson_command
+    src_root = b.environment.source_dir
+    bld_root = b.environment.build_dir
     priv_dir = os.path.join(bld_root, 'meson-private')
     dist_sub = os.path.join(bld_root, 'meson-dist')
 
-    buildfile = os.path.join(priv_dir, 'build.dat')
-
-    build = pickle.load(open(buildfile, 'rb'))
+    dist_name = b.project_name + '-' + b.project_version
 
-    dist_name = build.project_name + '-' + build.project_version
+    archives = determine_archives_to_generate(options)
 
     _git = os.path.join(src_root, '.git')
     if os.path.isdir(_git) or os.path.isfile(_git):
-        names = create_dist_git(dist_name, src_root, bld_root, dist_sub, build.dist_scripts)
+        names = create_dist_git(dist_name, archives, src_root, bld_root, dist_sub, b.dist_scripts)
     elif os.path.isdir(os.path.join(src_root, '.hg')):
-        names = create_dist_hg(dist_name, src_root, bld_root, dist_sub, build.dist_scripts)
+        names = create_dist_hg(dist_name, archives, src_root, bld_root, dist_sub, b.dist_scripts)
     else:
         print('Dist currently only works with Git or Mercurial repos')
         return 1
     if names is None:
         return 1
     error_count = 0
-    for name in names:
-        rc = check_dist(name, meson_command, priv_dir) # Check only one.
-        if rc == 0:
+     # Check only one.
```

Comment:
__[Flake8]__ [E114] indentation is not a multiple of four (comment) ([link](https://sider.review/gh/repos/19784232/pulls/5694#issue-4359067))

<sub>posted by Sider</sub>